api/v1alpha1/openapi.yaml

```diff
@@ -1307,6 +1307,390 @@ paths:
+      # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L2705
+        - name: name
+          # TODO - k8s does not have this parameter - keep?
+          in: path
+          description: name of the CertificateSigningRequest
+          required: true
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CertificateSigningRequest'
+        required: true
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequest'
+        "201":
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequest'
+        "401":
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "404": # TODO - k8s does not have this
+          description: NotFound
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /api/v1/certificatesigningrequests/{name}/approval:
+  # TODO - k8s has a patch method to partially update - do we want this?
+  # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L2302
+    put:
+      # TODO - k8s has dryRun, fieldManager, fieldValidation parameters
+      # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L2426
+      description: replace approval of the specified CertificateSigningRequest
+      operationId: replaceCertificateSigningRequestApproval
+      parameters:
+        - name: name
+          in: path
+          description: name of the CertificateSigningRequest
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequest'
+        "201":
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequest'
+        "401":
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    post:
+    # TODO - no equivalent for this in k8s - not sure how they initially create the approval request - is that handled through PUT?
```

Comment:
Yes. `certificatesigningrequests/{name}/approval` is a sub-resource and at that point the `certificatesigningrequests/{name}` resource already exists and is just updated. Having `/approval` allows different RBAC rules to apply.

Reply:
Interesting. I'll remove this POST and just support PUT to mirror what they are doing.